easybuild/tools/systemtools.py

```diff
@@ -219,30 +222,40 @@ def get_cpu_speed():
         try:
```

Comment:
I don't think the BSD one takes into account cpuset and all that crap, the affinity one does, so that should be tried first (and it should always work on Linux)